llarp/rpc/lokid_rpc_client.cpp

```diff
@@ -0,0 +1,200 @@
+#include <rpc/lokid_rpc_client.hpp>
+
+#include <util/logging/logger.hpp>
+
+#include <router/abstractrouter.hpp>
+
+#include <nlohmann/json.hpp>
+
+#include <util/time.hpp>
+#include <util/thread/logic.hpp>
+
+namespace llarp
+{
+  namespace rpc
+  {
+    static lokimq::LogLevel
+    toLokiMQLogLevel(llarp::LogLevel level)
+    {
+      switch (level)
+      {
+        case eLogError:
+          return lokimq::LogLevel::error;
+        case eLogWarn:
+          return lokimq::LogLevel::warn;
+        case eLogInfo:
+          return lokimq::LogLevel::info;
+        case eLogDebug:
+          return lokimq::LogLevel::debug;
+        case eLogNone:
+        default:
+          return lokimq::LogLevel::trace;
+      }
+    }
+
+    LokidRpcClient::LokidRpcClient(LMQ_ptr lmq, AbstractRouter* r)
+        : m_lokiMQ(std::move(lmq)), m_Router(r)
+    {
+      // m_lokiMQ->log_level(toLokiMQLogLevel(LogLevel::Instance().curLevel));
+    }
+
+    void
+    LokidRpcClient::ConnectAsync(lokimq::address url)
+    {
+      LogInfo("connecting to lokid via LMQ at ", url);
+      m_lokiMQ->connect_remote(
+          url.zmq_address(),
```

Comment:
If you pull the latest lokimq master I fixed this API oversight by letting you pass in the `address` here.  (Which also potentially carries more info than just the zmq address -- for example, someone can use an address that specified curve encryption + server pubkey).